common/src/lib.rs

```diff
@@ -190,6 +209,21 @@ pub fn set_code(code_hash: H256, code: &[u8]) {
     sp_io::storage::set(&code_key(code_hash).0, code)
 }
 
+#[derive(Decode, Encode)]
+pub enum ProgramState {
+    Uninitialized { message_id: H256 },
```

Comment:
Would you elaborate by commenting each variant what it means for a program to be in some particular state?

Reply:
ok